features/netsocket/TCPSocket.cpp

```diff
@@ -28,17 +29,26 @@ TCPSocket::TCPSocket(TCPSocket *parent, nsapi_socket_t socket, SocketAddress add
     _stack = parent->_stack;
     _factory_allocated = true;
     _remote_peer = address;
-
+    _socket_stats.stats_new_socket_entry(this);
     _event = mbed::Callback<void()>(this, &TCPSocket::event);
     _stack->socket_attach(socket, &mbed::Callback<void()>::thunk, &_event);
 }
 
 TCPSocket::~TCPSocket()
 {
+    _socket_stats.stats_update_socket_state(this, SOCK_CLOSED);
+}
+
+nsapi_error_t TCPSocket::close()
+{
+    _socket_stats.stats_update_socket_state(this, SOCK_CLOSED);
+    return InternetSocket::close();
```

Comment:
`_socket_stats.stats_update_socket_state(this, SOCK_CLOSED);` can be called in the `InternetSocket::close()` as it is same for all socket types.

Therefore we don't need this `TCPSocket::close()` method.

Reply:
I do not have socket instance (this) in InternetSocket, hence cannot move `stats_update_socket_state` calls to InternetSocket.. 

Same is true for open and entry functions